### PR TITLE
Allow to serialize DistributionFactory

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

**Comment by jschueller** [comment]
cc @aurelieladier, due in 1.9

File: lib/src/Uncertainty/Model/DistributionFactoryImplementation.cxx [hunk 1]
```diff
@@ -18,29 +18,22 @@
...
-#include <cstdlib>
-#include <iomanip>
-#include <fstream>
+
 #include "openturns/DistributionFactoryImplementation.hxx"
-#include "openturns/Description.hxx"
-#include "openturns/Path.hxx"
-#include "openturns/Exception.hxx"
-#include "openturns/OTconfig.hxx"
-#include "openturns/Log.hxx"
```
Comment (dbarbier) [inline lib/src/Uncertainty/Model/DistributionFactoryImplementation.cxx]: Log.hxx and Exception.hxx are used in this file, they should not be removed

Comment (jschueller) [inline lib/src/Uncertainty/Model/DistributionFactoryImplementation.cxx]: ok for Exceptions but Log ?

**Comment by sofianehaddad** [comment]
hmm why serializing factories?

**Comment by jschueller** [comment]
We need to serialize an object that has factories as attributes.

**Comment by sofianehaddad** [comment]
ok

**Pull request merged**